packages/core/src/Tolgee.ts

```diff
@@ -43,20 +24,25 @@ export class Tolgee {
     return this.dependencyService.translationService;
   }
 
-  get coreService() {
-    return this.dependencyService.coreService;
-  }
-
   public get lang() {
     return this.properties.currentLanguage;
   }
 
-  public set lang(value) {
-    this.properties.currentLanguage = value;
-    (
-      this.dependencyService.eventService
-        .LANGUAGE_CHANGED as EventEmitterImpl<any>
-    ).emit(value);
+  /**
+   * This sets a new language.
+   *
+   * Using this setter can behave buggy when you change languages
+   * too fast, since it changes the language property before
+   * translations are actually loaded.
+   * @Deprecated use asynchronous changeLanguage method.
+   */
+  public set lang(newLanguage) {
+    this.properties.currentLanguage = newLanguage;
+    this.emitLangChangeEvent(newLanguage);
+
+    this.translationService.loadTranslations(newLanguage).then(() => {
+      this.emitLangChangeAndLoad(newLanguage);
```

Comment:
Couldn't we just reuse languageLoaded? It feels overkill to have 3 kinds of events ...